xmtp_mls/src/groups/message_history.rs

```diff
@@ -43,27 +60,78 @@ where
         };
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub(crate) async fn prepare_messages_to_send(
+        &self,
+    ) -> Result<Vec<StoredGroupMessage>, StorageError> {
+        println!("bundle_messages_to_send called");
+        let conn = self.store.conn()?;
+        let groups = conn.find_groups(None, None, None, None)?;
+        let mut all_messages: Vec<StoredGroupMessage> = vec![];
+
+        for StoredGroup { id, .. } in groups.clone() {
+            let messages = conn.get_group_messages(id, None, None, None, None, None)?;
+            println!("{:#?}", messages);
+            all_messages.extend(messages);
+        }
+
+        println!("groups: {:#?}", groups);
+        println!("# of group messages: {:?}", all_messages.len());
+        Ok(all_messages)
+    }
 }
 
-pub(crate) fn new_message_history_request() -> MessageHistoryRequest {
-    MessageHistoryRequest {
-        pin_code: new_pin(),
-        request_id: new_request_id(),
+struct HistoryRequest {
+    pin_code: String,
+    request_id: String,
+}
+
+impl HistoryRequest {
+    pub(crate) fn new() -> Self {
+        Self {
+            pin_code: new_pin(),
+            request_id: new_request_id(),
+        }
     }
 }
 
-#[allow(dead_code)]
-pub(crate) fn new_message_history_reply(
-    id: &str,
-    url: &str,
-    hash: Vec<u8>,
-    exp: i64,
-) -> MessageHistoryReply {
-    MessageHistoryReply {
-        request_id: id.into(),
-        backup_url: url.into(),
-        backup_file_hash: hash,
-        expiration_time_ns: exp,
+impl From<HistoryRequest> for MessageHistoryRequest {
+    fn from(req: HistoryRequest) -> Self {
+        MessageHistoryRequest {
+            pin_code: req.pin_code,
+            request_id: req.request_id,
+        }
+    }
+}
+
+struct HistoryReply {
+    request_id: String,
```

Comment:
Doesn't this struct need to have some sort of encryption keys in it?

Reply:
Yep.  I was going to put that change in the proto first, and once that's in, make the corresponding addition here, too.

Reply:
👍

Reply:
https://github.com/xmtp/proto/pull/179